File: Problems/0424-longest-repeating-character-replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        cnt = [0] * 26
        left = 0
        best = 0
        ans = 0

        for right in range(len(s)):
            idx = ord(s[right]) - 65
            cnt[idx] += 1

            if cnt[idx] > best:
                best = cnt[idx]

            while right - left + 1 - best > k:
                cnt[ord(s[left]) - 65] -= 1
                left += 1

            if right - left + 1 > ans:
                ans = right - left + 1

        return ans

File: Problems/0424-longest-repeating-character-replacement.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    int characterReplacement(string s, int k) {
        int cnt[26] = {};
        int left = 0, best = 0, ans = 0;

        for (int right = 0; right < s.size(); right++) {
            int idx = s[right] - 'A';
            best = max(best, ++cnt[idx]);

            while (right - left + 1 - best > k)
                cnt[s[left++] - 'A']--;

            ans = max(ans, right - left + 1);
        }

        return ans;
    }
};